tasks/Functions-client/setAutoRequest.js

```diff
@@ -44,92 +50,74 @@ const setAutoRequest = async (contract, taskArgs) => {
     throw Error("Gas limit must be less than or equal to 300,000")
   }
 
-  console.log(`Setting the Functions request in AutomatedFunctionsConsumer contract ${contract} on ${network.name}`)
+  console.log(`\nSetting the Functions request in AutomatedFunctionsConsumer contract ${contract} on ${network.name}`)
 
   const autoClientContractFactory = await ethers.getContractFactory("AutomatedFunctionsConsumer")
   const autoClientContract = await autoClientContractFactory.attach(contract)
 
   const unvalidatedRequestConfig = require(path.isAbsolute(taskArgs.configpath)
     ? taskArgs.configpath
     : path.join(process.cwd(), taskArgs.configpath))
+
   const requestConfig = getRequestConfig(unvalidatedRequestConfig)
 
-  // doGistCleanup indicates if an encrypted secrets Gist was created automatically and should be cleaned up by the user after use
-  let doGistCleanup = !(requestConfig.secretsURLs && requestConfig.secretsURLs.length > 0)
-  const request = await generateRequest(requestConfig, taskArgs)
+  console.log("\nEncrypting secrets and uploading to DON...")
+  const functionsRouterAddress = networks[network.name]["functionsRouter"]
+  const donId = networks[network.name]["donId"]
+  const signer = await ethers.getSigner()
 
-  if (doGistCleanup && request.secrets) {
-    console.log(
-      `Be sure to delete the Gist ${request.secretsURLs[0].slice(0, -4)} once encrypted secrets are no longer in use!\n`
-    )
-  }
+  const secretsManager = new SecretsManager({
+    signer,
+    functionsRouterAddress,
+    donId,
+  })
 
-  const functionsRequestBytes = await autoClientContract.generateRequest(
-    request.source,
-    request.secrets ?? [],
-    request.args ?? []
-  )
+  await secretsManager.initialize()
+  const encryptedSecretsObj = await secretsManager.buildEncryptedSecrets(requestConfig.secrets)
+  const slotId = parseInt(taskArgs.slotid)
+  const minutesUntilExpiration = 10 // Minimum 5 minutes supported.
 
-  const store = new RequestStore(hre.network.config.chainId, network.name, "automatedConsumer")
-  const previousSecretURLs = []
-  try {
-    const artifact = await store.read(taskArgs.contract)
-    if (artifact.activeManagedSecretsURLs) previousSecretURLs = artifact.secretsURLs
-  } catch {
-    /* new request, continue */
+  const { version, success } = await secretsManager.uploadEncryptedSecretsToDON({
+    encryptedSecretsHexstring: encryptedSecretsObj.encryptedSecrets,
+    gatewayUrls: networks[network.name]["gatewayUrls"],
+    storageSlotId: slotId,
+    minutesUntilExpiration,
+  })
+
+  if (!success) {
+    throw Error("\nFailed tp upload encrypted secrets to DON.")
```

Comment:
Note that `uploadEncryptedSecretsToDON` will throw a descriptive error if it fails for all nodes.  success == false just indicates that one of the nodes failed.  Not quite sure how best to communicate this to users though...

Reply:
Is this still accurate given the changes we noted when there are two gateway URLs being passed? I got a descriptive error when the second URL threw an error (which indicates first URL was fine, and success would == false)

https://chainlink-core.slack.com/archives/C04DBHV754K/p1694036838760739?thread_ts=1694036015.102869&cid=C04DBHV754K

Reply:
I checked the stack and the error I linked to arises from secretsManager.uploadEncryptedSecretsToDON() which means that even that it seems to be throwing an error even if one node succeeded

Reply:
Gotcha.  Bolek & I will discuss this and I may need to handle this better in the toolkit.